akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
```

Comment:
As I understand this:
- the LMF creates "lambdas" which are of the type of whatever the target type was, this is transparent by design
- however it also generates readResolve/readReplace/deserializeLambda magical java serialization methods which allow serializing those lambdas...
- those become https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/SerializedLambda.html

Still... rather magical. I assume every lambda magically is also serializable them?